Fix path translation when MSYS_NO_PATHCONV=1

luka8088 opened the pull request:
Under windows git bash the following produces an unexpected result:

```bash
$ MSYS_NO_PATHCONV=1 vendor/bin/some-package-script
Could not open input file: /c/Users/me/project/vendor/some-package-script/some-package-script
```

It is due to `MSYS_NO_PATHCONV` disabling automatic path translations.
This patches detects such cases and forces the path translation so that the example command works as expected.

Reference:
https://github.com/git-for-windows/msys2-runtime/pull/11

Seldaek commented:
Thanks but please rebase/re-target this PR to the 1.10 branch so it can be included as a bugfix in the next patch release.

johnstevenson commented:
> Under windows git bash the following produces an unexpected result:

Far from being unexpected, this is actually expected NOT to work, so I'm not entirely sure this is Composer's responsibility.

https://github.com/git-for-windows/git/issues/407#issuecomment-141548288

As I understand it, using `MSYS_NO_PATHCONV` is to solve a specific problem and it should only be set temporarily.

Seldaek commented:
Yup I agree this does make it sound like patching this is rather unnecessary.

luka8088 commented:
> Thanks but please rebase/re-target this PR to the 1.10 branch so it can be included as a bugfix in the next patch release.

Rebased

luka8088 commented:
> 
> 
> > Under windows git bash the following produces an unexpected result:
> 
> Far from being unexpected, this is actually expected NOT to work, so I'm not entirely sure this is Composer's responsibility.
> 
> [git-for-windows/git#407 (comment)](https://github.com/git-for-windows/git/issues/407#issuecomment-141548288)
> 
> As I understand it, using `MSYS_NO_PATHCONV` is to solve a specific problem and it should only be set temporarily.

Interesting, I didn't know they had that attitude towards it. I have it set as a default all the time because as a practical matter addressing the issues that it causes is much less work than using it selectively.

I do understand if you prefer not to merge it though.

johnstevenson commented:
> I do understand if you prefer not to merge it though.

While its certainly not my call, it is interesting that Git for Windows doesn't even try to "fix" things for its own known `gitk` invocation (as per the thread I linked to).

There are other ways to get around the path conversion: http://www.mingw.org/wiki/Posix_path_conversion

Also from https://github.com/git-for-windows/build-extra/blob/master/ReleaseNotes.md 

> Alternatively, you can double the first slash to avoid POSIX-to-Windows path conversion, e.g. "//usr/bin/bash.exe".

Although it all looks very cumbersome!

Seldaek commented:
Yeah, overall this seems like it's not really worth a patch.

Besides, MSYS/cygwin are really best replaced by WSL IMO at this point, which offers a `wslpath` for cross-boundary scripting when absolutely needed.

luka8088 commented:
> 
> 
> > I do understand if you prefer not to merge it though.
> 
> While its certainly not my call, it is interesting that Git for Windows doesn't even try to "fix" things for its own known `gitk` invocation (as per the thread I linked to).
> 
> There are other ways to get around the path conversion: http://www.mingw.org/wiki/Posix_path_conversion
> 
> Also from https://github.com/git-for-windows/build-extra/blob/master/ReleaseNotes.md
> 
> > Alternatively, you can double the first slash to avoid POSIX-to-Windows path conversion, e.g. "//usr/bin/bash.exe".
> 
> Although it all looks very cumbersome!

Yeah, I read already through that but the thing is I have lots of cross-platform bash scripts, thanks anyways though!

luka8088 commented:
> 
> 
> Yeah, overall this seems like it's not really worth a patch.
> 
> Besides, MSYS/cygwin are really best replaced by WSL IMO at this point, which offers a `wslpath` for cross-boundary scripting when absolutely needed.

Yeah, maybe it's time that I look into migrating onto WSL, thanks!
